design/ceph/osd-migration.md

```diff
@@ -0,0 +1,239 @@
+# OSD Backend Store Migration
+
+## SUMMARY
+
+- Configure OSD backend store via `cephCluster` CR so that new OSDs can use it.
+- Existing OSDs should be able to migrate to use a new backend store configured via `cephCluster` CR. The migration process should include destroying the existing OSDs one by one, wiping the disk and then creating a new OSD on that disk.
+
+## MOTIVATION:
+
+- Ceph uses BlueStore as a special-purpose storage back end designed specifically for managing data on disk for Ceph OSD workloads. Ceph team is working on improving this backend storage that will help in significantly improving the performance, especially in disaster recovery scenarios. This new OSD backend is known as `bluestore-rdr`. In the future, other backends will also need to be supported, such as `seastore`.
+- To support any new and improved storage backend, the existing OSD needs to be destroyed, disks should be wiped out and new OSD should be created using the latest storage backend.
+
+## GOALS
+
+- Newly provisioned clusters should be able to use the specific storage backend while creating OSDs.
+- Existing clusters should be able to migrate the OSDs to use a specific storage backend, without downtime and without risking data loss.
+
+## PROPOSAL
+
+- Add a new field called `spec.Storage.osdStore` in the ceph cluster yaml.
+
+```yaml
+storage:
+  osdStore:
```

Comment:
Actually, I see in types.go that we already have a `store` with a `type` setting. So we can use the existing type and just add the new `updateStore` setting to the [OSDStore struct](https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/types.go#L2595)?

Reply:
Yes, that's part of the same effort. The existing `OSDStore` that's there in master was for the Greenfield case. We had decided to keep the same design PR for both Greenfield and Brownfield scenarios.

Reply:
Oh of course, good to know it's for the same effort. :) 
Then I believe the yaml example here and in below paragraphs need to be renamed, right? osdStore --> store

Reply:
ah right. My bad. Its `store`.  Update the design doc.

Reply:
marking it as resolved.